### PR TITLE
Added template and arguments to support ACI CNI plugin

### PR DESCRIPTION
**Pull request by Bhavanaashok33**
Introducing ACI CNI network provider 
ACI CNI supports rke version 1.2.4+ 
Tested the code changes on the ACI fabric 

Signed-off-by: bashokba <bashokba@cisco.com>

### CODE REVIEW

File: rke/schema_rke_cluster_system_images.go [hunk 1]
```diff
@@ -132,6 +132,30 @@ func rkeClusterSystemImagesFields() map[string]*schema.Schema {
...
+			Optional: true,
+		},
+		"aci_ovs_container": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"aci_controller_container": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
```
Comment (rawmind0) [inline rke/schema_rke_cluster_system_images.go:158]: Checking [rke system images](https://github.com/rancher/rke/blob/release/v1.2/types/rke_types.go#L189), are `AciGbpServerContainer` and `AciOpflexServerContainer ` not required as configurable images??

**Review by rawmind0 — COMMENTED** [review]
@Bhavanaashok33 , thanks for your PR.

Could you please take a look to review comments?? Also, new aci arguments should be documented on `docs/resources/cluster.md`, could you please add them to the PR??

File: rke/schema_rke_cluster_system_images.go (hunk 1 above)
Comment (Bhavanaashok33) [reply rke/schema_rke_cluster_system_images.go:158]: These images are to support the overlay case. On rke upstream, the validator doesn't check for these images yet. After we push the changes to support overlay case for ACI CNI on rke side, I will incorporate the changes to include AciGbpServerContainer and AciOpflexServerContainer on terraform.

File: rke/structure_rke_cluster_system_images.go [hunk 2]
```diff
@@ -40,6 +40,12 @@ func flattenRKEClusterSystemImages(in rancher.RKESystemImages) []interface{} {
 	obj["metrics_server"] = in.MetricsServer
 	obj["windows_pod_infra_container"] = in.WindowsPodInfraContainer
 	obj["nodelocal"] = in.Nodelocal
+	obj["aci_cni_deploy_container"] = in.AciCniDeployContainer
+	obj["aci_host_container"] = in.AciHostContainer
+	obj["aci_opflex_container"] = in.AciOpflexContainer
+	obj["aci_mcast_container"] = in.AciMcastContainer
+	obj["aci_ovs_container"] = in.AciOpenvSwitchContainer
+	obj["aci_controller_container"] = in.AciControllerContainer
```
[the remark this answers is not in the document]
Comment (Bhavanaashok33) [reply rke/structure_rke_cluster_system_images.go:49]: Have addressed the comment below.

File: rke/schema_rke_cluster_network.go [hunk 3]
```diff
@@ -69,6 +71,114 @@ func rkeClusterNetworkWeaveFields() map[string]*schema.Schema {
...
+		"apic_hosts": {
+			Type:     schema.TypeList,
+			Required: true,
+			Elem: &schema.Schema{
+				Type: schema.TypeString,
+			},
+		},
+		"token": {
+			Type:     schema.TypeString,
+			Required: true,
```
[the remark this answers is not in the document]
Comment (Bhavanaashok33) [reply rke/schema_rke_cluster_network.go:89]: Yeah sure, let's not expose it.

File: rke/schema_rke_cluster_network.go [hunk 4]
```diff
@@ -69,6 +71,114 @@ func rkeClusterNetworkWeaveFields() map[string]*schema.Schema {
...
+		},
+		"apic_user_name": {
+			Type:     schema.TypeString,
+			Required: true,
+		},
+		"apic_user_key": {
+			Type:     schema.TypeString,
+			Required: true,
```
[the remark this answers is not in the document]
Comment (Bhavanaashok33) [reply rke/schema_rke_cluster_network.go:97]: sure, make sense.

**Review by rawmind0 — APPROVED** [review]
Thanks for the update @Bhavanaashok33 

LGTM

**Pull request merged**